resource-core/src/main/java/org/jumpmind/metl/core/runtime/resource/aws/S3.java

```diff
@@ -0,0 +1,111 @@
+package org.jumpmind.metl.core.runtime.resource.aws;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.jumpmind.metl.core.runtime.MisconfiguredException;
+import org.jumpmind.metl.core.runtime.resource.AbstractResourceRuntime;
+import org.jumpmind.properties.TypedProperties;
+
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProviderChain;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+
+public class S3 extends AbstractResourceRuntime {
+    public static final class Settings {
+        /** Name of the setting that identifies the AWS S3 region. */
+        public static final String REGION = "aws.s3.region";
+
+        public static final String USE_REGION_AS_CRYPTO_DEFAULT = "aws.s3.region.as.crypto.default";
+
+        public static final String BUCKET_NAME = "aws.s3.bucket.name";
+
+        Settings() {
+            throw new UnsupportedOperationException("do not instantiate");
+        }
+    }
+
+    private final AtomicReference<S3Directory> bucketReference = new AtomicReference<>();
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <T> T reference() {
+        return (T) bucketReference.get();
+    }
+
+    @Override
+    protected void start(final TypedProperties properties) {
+        super.start(properties);
+
+        bucketReference.set(createS3Directory(properties));
+    }
+
+    private S3Directory createS3Directory(final TypedProperties properties) {
+        String regionId = properties.get(S3.Settings.REGION);
+        Region region = (regionId != null) ? Region.of(regionId) : null;
+
+        String bucketName = properties.get(S3.Settings.BUCKET_NAME, "").trim();
+        if (bucketName == null || bucketName.isEmpty()) {
+            throw new MisconfiguredException(S3.Settings.BUCKET_NAME);
+        }
+
+        ClientSideCrypto crypto = properties.is(ClientSideCrypto.Settings.ENABLED, false)
+                ? createClientSideCrypto(properties, regionId)
+                : null;
+
+        String listFilesDelimiter = properties.get(S3Directory.Settings.LIST_FILES_DELIMITER,
+                S3Directory.Settings.DEFAULT_LIST_FILES_DELIMITER);
+        int transferWindowSize = properties.getInt(S3Directory.Settings.TRANSFER_WINDOW_SIZE,
+                S3Directory.Settings.DEFAULT_TRANSFER_WINDOW_SIZE);
+
+        return new S3Directory(getResource(), region, bucketName, crypto,
```

Comment:
The purpose of getResource() here is unclear to me. What is it supposed to be?  I think I may need to alter how this initialization is performed.